src/python/pants/backend/python/lint/bandit/rules.py

```diff
@@ -117,19 +122,37 @@ async def bandit_lint_partition(
     address_references = ", ".join(
         sorted(field_set.address.reference() for field_set in partition.field_sets)
     )
-
+    output_dir = lint_options.output_dir
+    output_path = output_dir / "bandit_report.txt" if output_dir else None
+    bandit_args = generate_args(
+        specified_source_files=specified_source_files,
+        bandit=bandit,
+        output_file=output_path.name if output_path else None,
+    )
     process = requirements_pex.create_process(
         python_setup=python_setup,
         subprocess_encoding_environment=subprocess_encoding_environment,
         pex_path="./bandit.pex",
-        pex_args=generate_args(specified_source_files=specified_source_files, bandit=bandit),
+        pex_args=bandit_args,
+        output_files=(output_path.name,) if output_path else None,
         input_digest=input_digest,
         description=(
             f"Run Bandit on {pluralize(len(partition.field_sets), 'target')}: {address_references}."
         ),
     )
     result = await Get(FallibleProcessResult, Process, process)
-    return LintResult.from_fallible_process_result(result, linter_name="Bandit")
+    results_file = None
+    if output_path:
+        report_file_snapshot = await Get(
+            Snapshot, SnapshotSubset(result.output_digest, PathGlobs([output_path.name]))
+        )
+        if report_file_snapshot.is_empty or len(report_file_snapshot.files) != 1:
+            raise Exception(f"Unexpected report file snapshot: {report_file_snapshot}")
```

Comment:
I think it would be more descriptive to have two distinct exceptions. One says "Report file is empty" and the other says "More than one report found: {snapshot.files}". Or something like that.

Reply:
Ah, ditto on improving this for Flake8 as well. It might be worth factoring up some of this functionality, maybe into a new util file under `backend/python/lint/`? Functionality that's specific to Python linters.

Reply:
I really don't expect this code to be hit, considering what we pass as `PathGlobs` however, the way the API works, I feel making this check/assertion here is prudent to avoid a confusing error at a later stage.
so splitting this into two district errors seems excessive.
I'll change the log to include more information.

Reply:
Okay. At a minimum, outputting `snapshot.files` would be better than `snapshot`. There's no need for the end user to see the `snapshot.digest` in the output, for example.